core/src/mindustry/entities/comp/PayloadComp.java

```diff
@@ -162,4 +162,34 @@ void contentInfo(Table table, float itemSize, float width){
             table.image(p.icon(Cicon.small)).size(itemSize).padRight(pad);
         }
     }
+
+    @Override
+    public void destroy(){
+        //if this unit was carrying other units, spawn them but at reduced health
+        payloads.each(payload -> {
+            if(payload instanceof UnitPayload p){
+                Unit u = p.unit;
+
+                //drop all units, let ground units die on walls
+                Fx.unitDrop.at(this);
+
+                if(Vars.net.client()){
+                    //clear removed state of unit so it can be synced
+                    Vars.netClient.clearRemovedEntity(p.unit.id);
+                }else{
+                    //clients do not drop payloads
+                    u.set(this);
+                    u.trns(Tmp.v1.rnd(Mathf.random(2f)));
+                    u.rotation(rotation);
+                    //reset the ID to a new value to make sure it's synced
+                    u.id = EntityGroup.nextId();
+                    //decrement count to prevent double increment
+                    if(!u.isAdded()) u.team.data().updateCount(u.type, -1);
+                    //take damage from the landing
+                    u.health -= u.maxHealth * (1 - u.type.crashDurability);
```

Comment:
```suggestion
                    u.health -= 200 + u.maxHealth * (1 - u.type.crashDurability)
```
Instead of the 0 crash durability, there should be some damage that is independent of maxHealth. The 200 should probably be set in a constant somewhere, but I dont' know where.

Reply:
Maybe rename `crashDurability` and instead make it just "damage taken on crash?"

Reply:
Idea - Upon crash, units take crashDurability% damage, plus the damage value for the payload unit crashing onto enemies, plus unit explosion damage.

Reply:
I feel like it should have a different name so that it doesn't get confused with the `crashDamageMultiplier` for just flying units dying.

Reply:
Sorry, I thought something was different. When I suggested, I meant that it should look up the damage for the carrying unit dying/crashing/exploding, and apply that to the dropped units.